ovirt/fakeovirt_deployment.yml

```diff
@@ -24,6 +24,16 @@ spec:
             value: konveyor-forklift
           - name: PORT
             value: "12346"
+      - name: imageio
+        image: localhost:5001/imageio:latest
+        ports:
+        - containerPort: 12345
+        command: ["/bin/bash"]
+        args:
+        - -c
+        - ovirt-imageio& sleep 15 && 
+          ovirt-imageioctl add-ticket myticket.json &&
+          tail -n+1 -f /var/log/ovirt-imageio/daemon.log
```

Comment:
This replicates what you can achieve with the containerized imageio server that is provided at https://quay.io/repository/ovirt/ovirt-imageio.

This container serves an image, automatically generating and using a ticket, at the http port 80 (non-secure connection).

```
podman run \                           
        --interactive \
        --tty \
        --rm \
        quay.io/ovirt/ovirt-imageio:latest --help
usage: entrypoint.py [-h] [--ticket-id TICKET_ID] image

ovirt-imageio

positional arguments:
  image                 Path to image.

options:
  -h, --help            show this help message and exit
  --ticket-id TICKET_ID
                        Optional. Set the ID of the ticket. It is recommended to use with random string for better security. Defaults to the image filename.
```

It mostly only takes the path (in the container) to the image that we want to serve, and allows specifying a ticket id.

So it could be something like (untested, just an approximate example):
```
      - name: imageio
        image: quay.io:ovirt/ovirt-imageio:latest
        ports:
        - containerPort: 12345:80
        volumes:
        - ovirt-imageio/cirros.raw:/images/cirros.raw
        command: ["/images/cirros.raw", "--ticket-id", "cirros"]
```
To be more clear, this is how it would look like in the command line:
```
$ podman run \                           
        --interactive \
        --tty \
        --rm \
        --publish 12345:80 \
        --volume ovirt-imageio/:/images/:Z \
        quay.io/ovirt/ovirt-imageio:latest /images/cirros.raw --ticket-id cirros
```
Which serves the image cirros.raw at `http://localhost:12345/images/cirros`.
At the moment, we only support non-secure connections. We are also adding `--insecure` option to `ovirt-img` to nicely support this option from both server and client side.

Or maybe another option would be to build another image FROM the ovirt-imageio container, and mount and run the image inside it, instead of using it directly here.

In any case, it would be great if you could try to integrate this containerized server, rather than replicating efforts. If there is any proposal for improvement or need for more configuration options, we can work those out and improve the container.

Reply:
Thanks a lot for the examples!  I have chosen to build the image with the cirros.
Can you please check out it out again if you have some comments?